main.go

```diff
@@ -79,23 +79,20 @@ func main() {
 
 	sources := source.NewMultiSource(source.LookupMultiple(cfg.Sources...)...)
 
-	googleProvider, err := dnsprovider.NewGoogleProvider(cfg.GoogleProject, cfg.DryRun)
-	if err != nil {
-		log.Fatal(err)
-	}
+	dnsprovider.Register("google", dnsprovider.NewGoogleProvider(cfg.GoogleProject, cfg.DryRun))
+	dnsprovider.Register("aws", dnsprovider.NewAWSProvider(cfg.DryRun))
```

Comment:
Would it not make more sense to only register/initialize the providers which are required?

E.g. the interface would be more like this:

```go
provider, err := dnsprovider.Initialize(cfg.DNSProvider)
```


*Disregard if I'm missing some context.*